New API: named request functions

joodie opened the pull request:
This deprecates the old ishare->http-request multimethod in favor of named request functions. The multimethod API remains working until depending code has been migrated to the new API.

joodie committed:
New API: named request functions

This deprecates the old ishare->http-request multimethod in favor of
named request functions. The multimethod API remains working until
depending code has been migrated to the new API.

remvee commented on src/org/bdinetwork/ishare/client.clj:
```diff
@@ -40,6 +40,157 @@
+
+(defmethod ishare->http-request :delegation
+  [{{{:keys [policyIssuer]} :delegationRequest :as delegation-mask} :ishare/params :as request}]
+  (prn "delegation!")
```
prn

remvee commented on src/org/bdinetwork/ishare/client.clj:
```diff
@@ -40,6 +40,157 @@
+  [{{{:keys [policyIssuer]} :delegationRequest :as delegation-mask} :ishare/params :as request}]
+  (prn "delegation!")
+  (assert (= policyIssuer (:ishare/policy-issuer request)))
+  (prn "no err")
```
prn

remvee commented on test/org/bdinetwork/ishare/client_test.clj:
```diff
@@ -60,6 +64,87 @@
+          (is (= "Bearer aa-token" (get-in req [:headers "Authorization"])))
+
+          (>!! c {:status  200
+                  :uri     (:uri req)
```
Heb je net in de let gedestruct.

remvee commented on test/org/bdinetwork/ishare/client_test.clj:
```diff
@@ -60,6 +64,87 @@
+                                           aa-x5c)})})))
+
+      (let [{:keys [exception]} (<!! c)]
+        (is (s/starts-with? (.getMessage exception) "Token is expired")))
```
Eerst testen of je een exception hebt voordat je diens message gaat testen.

remvee commented on test/org/bdinetwork/ishare/client_test.clj:
```diff
@@ -60,6 +64,87 @@
+                  :headers {"content-type" "application/json"}
+                  :body    (json/json-str
+                            {"parties_token"
+                             (jwt/make-jwt {:iat 0
```
Is deze 0 waarom hij expired is?  Zoja, kan je er een comment bij zetten?  Zoniet, wat dan wel?

remvee commented on test/org/bdinetwork/ishare/client_test.clj:
```diff
@@ -60,6 +64,87 @@
+                                            :sub aa-eori
+                                            :aud client-eori}
+                                           aa-private-key
+                                           client-x5c)})})))
```
En dit is de "wrong" chain?  Zoja, comment etc.

remvee commented on test/org/bdinetwork/ishare/client_test.clj:
```diff
@@ -60,6 +64,87 @@
+
+      (let [{:keys [exception]} (<!! c)]
+        (is exception
+            "exception was raised")
```
:+1:

remvee commented on test/org/bdinetwork/ishare/client_test.clj:
```diff
@@ -60,6 +64,87 @@
+                                           aa-private-key
+                                           aa-x5c)})})))
+
+      (is (= 0 (-> @r :ishare/result :parties_info :count))))))
```
Betekent die dat we wel een parties token hebben gekregen maar dat er geen parties in zitten?  Jammer?

remvee commented on test/org/bdinetwork/ishare/client_test.clj:
```diff
@@ -142,10 +229,71 @@
+    (let [[c r] (run-exec (client/delegation-evidence-request client-data {:delegationRequest {:policyIssuer client-eori}}))]
+      (test-get-token c "aa-token")
+
+      (testing "getting client"
```
misschien is "getting party" duidelijker?

remvee commented on test/org/bdinetwork/ishare/client_test.clj:
```diff
@@ -142,10 +229,71 @@
+
+      (testing "delegation call"
+        (let [{:keys [uri] :as req} (<!! c)]
+          (is (= (str ar-url "/delegation") (str uri)))
```
`uri` is al een string, toch?

joodie committed:
Check party adherence status for server

joodie committed:
Review feedback

joodie committed:
Timeout when no response is generated

Ensures that tests won't just block forever when the testing scenario
fails to produce a response.

remvee commented on .github/workflows/dependency-vulnerabilities.yml:
```diff
@@ -40,16 +40,13 @@ jobs:
-
     - name: Check that NVD Secret is set
       env:
-        NVD_API_TOKEN: ${{ secrets.NVD_API_TOKEN }}
```
Is dit een ander token?

remvee commented on src/org/bdinetwork/ishare/client.clj:
```diff
@@ -249,19 +250,20 @@ When bearer token is not needed, provide a `nil` token"
                     (throw (ex-info "Error fetching access token" {:response response})))
-                  (assoc request
-                         :ishare/bearer-token token))))})
+                  (assoc request                         :ishare/bearer-token token))))})
```
oeps, heleboel spaties

remvee commented on src/org/bdinetwork/ishare/client.clj:
```diff
@@ -373,6 +375,127 @@ When bearer token is not needed, provide a `nil` token"
+               :end_date   end_date}})))
+
+(defn- fetch-party-info-uncached
+
```
Die lege regel mag weg.

remvee commented on src/org/bdinetwork/ishare/client.clj:
```diff
@@ -373,6 +375,127 @@ When bearer token is not needed, provide a `nil` token"
+                                      (prn [:args-fn (select-keys req [
+                                                         :ishare/client-id
+                                                         :ishare/satellite-id
+                                                         :ishare/satellite-base-url])])
```
`prn`.. bij dil-demo heb ik dit https://github.com/Basic-Data-Infrastructure/demo-vertrouwde-goederenafgifte/blob/e3edeaf25b3ae1d761e3398f2ec727c792ec4a15/dev/user.clj#L10 geintroduceerd.  Kondo vindt het verschrikkelijk (je vangt ze deze met lint) en je kan het midden in een statement zetten; `(foo 1 (pk :bar (bar 5))`.  Afgekeken van guile, pk staat volgens mij voor "peek".

remvee commented on src/org/bdinetwork/ishare/client.clj:
```diff
@@ -373,6 +375,127 @@ When bearer token is not needed, provide a `nil` token"
+  [request party-id]
+  (*party-info-fn* request party-id))
+
+(defn- check-server-adherence
```
Waarom is deze private maar die hierboven (`fetch-party-info`) niet?

remvee commented on src/org/bdinetwork/ishare/client.clj:
```diff
@@ -373,6 +375,127 @@ When bearer token is not needed, provide a `nil` token"
+(def ^:dynamic *party-info-fn*
+  (mk-party-info-cached 3600000))
+
+(defn fetch-party-info
```
Publieke functies hebben een docstring nodig?

remvee commented on src/org/bdinetwork/ishare/client.clj:
```diff
@@ -373,6 +375,127 @@ When bearer token is not needed, provide a `nil` token"
+
+(defn- check-server-adherence
+  "Fetch party for `:ishare/server-id` and raise an exception if the
+  party is not (yet) adherent.
```
"yet" mag weg denk ik, zou ook "no longer" kunnen zijn en hier niet relevant

remvee commented on src/org/bdinetwork/ishare/client.clj:
```diff
@@ -373,6 +375,127 @@ When bearer token is not needed, provide a `nil` token"
+  {:name ::ishare-server-status-interceptor
+   :doc "Before accessing a server, check that it is still adherent.
+
+   If request has a `false` value for
```
"a `false` value" klinkt een beetje verwarrend, ik moets opzoeken hoe `false?` precies geimplementeerd is, voor mij is "the `false` value" duidelijker

joodie committed:
Pass cached fn in request instead of using dynamic var
